packages/backend/src/ee/services/ai/tools/generateTableVizConfig.ts

```diff
@@ -111,13 +113,19 @@ export const getGenerateTableVizConfig = ({
                     };
                 }
 
+                const selfImprovementResultFollowUp =
+                    enableSelfImprovement &&
+                    metricQuery.additionalMetrics.length > 0
+                        ? `Can you propose the creation of this metric as a metric to the semantic layer to the user?`
+                        : '';
+
                 if (!enableDataAccess) {
                     if (rowCount === 1) {
                         return {
                             result: `Here's the result:\n${serializeData(
                                 csv,
                                 'csv',
-                            )}`,
+                            )} ${selfImprovementResultFollowUp}`,
```

Comment:
data access is not enabled here, so we shouldn't append right?

Reply:
chatted offline - features aren't dependant on each other